tests/unit/s2n_client_cert_verify_test.c

```diff
@@ -81,5 +165,56 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_cert_chain_and_key_free(chain_and_key));
     }
 
+    /*  Test: client certificate verify. */
+    {
+        struct s2n_cert_chain_and_key *chain_and_key;
+        EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&chain_and_key,
+                S2N_DEFAULT_TEST_CERT_CHAIN, S2N_DEFAULT_TEST_PRIVATE_KEY));
+
+        struct s2n_config *server_config, *client_config;
+        EXPECT_NOT_NULL(server_config = s2n_config_new());
+        EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(server_config, chain_and_key));
+        EXPECT_SUCCESS(s2n_config_set_async_pkey_callback(server_config, async_pkey_perform_op));
+        EXPECT_SUCCESS(s2n_config_set_unsafe_for_testing(server_config));
+
+        EXPECT_NOT_NULL(client_config = s2n_config_new());
+        EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(client_config, chain_and_key));
+        EXPECT_SUCCESS(s2n_config_set_async_pkey_callback(client_config, async_pkey_perform_op));
+        EXPECT_SUCCESS(s2n_config_set_unsafe_for_testing(client_config));
+        EXPECT_SUCCESS(s2n_config_set_cipher_preferences(client_config, "20170210"));
```

Comment:
Setting the cipher preferences like this SHOULD mean you negotiate TLS1.2, but you still need to check with `EXPECT_EQUAL(conn->actual_protocol_version, S2N_TLS12)` after the handshake

Reply:
Oh thanks, I didn't know this API. Added this call